imfilebrowser.h

```diff
@@ -90,8 +90,24 @@ namespace ImGui
         // default value is 0 (the first type filter)
         void SetCurrentTypeFilterIndex(int index);
 
+        // when ImGuiFileBrowserFlags_EnterNewFilename is set
+        // this function will pre-fill the input dialog with a filename.
+        void SetInputName(std::basic_string<char> input) noexcept
+        {
+            if((flags_ & ImGuiFileBrowserFlags_EnterNewFilename)) {
+                if (input.size() >= static_cast<size_t>(INPUT_NAME_BUF_SIZE)) {
+                    // if input doesn't fit trip off characters
+                    input.resize(INPUT_NAME_BUF_SIZE - 1);
+            }
+            std::copy(input.begin(), input.end(), inputNameBuf_->begin());
+            inputNameBuf_->at(input.size()) =
+                '\0'; // I think we need to add the null.
+            selectedFilenames_ = { inputNameBuf_->data() };
```

Comment:
this line doesn't fit the `noexcept` requirement. perhaps this function shouldn't be marked as `noexcept`. the standard's policy on `noexcept` is to mark functions that must not fail, instead of those specified not to throw

Reply:
removed noexcept